### DISCUSSION TITLE
tracedoc 或者 syncobj 在 skynet 中的应用

### DISCUSSION THREAD

**Discussion by hanxi** [cloudwu/skynet#1500] [Q&A]
- https://github.com/cloudwu/syncobj 这个只支持一层表
- https://github.com/cloudwu/tracedoc
- https://github.com/jojo59516/tracedoc 这个是上面的 fork，改了挺多实现的一个版本

这两个东西可以用于 skynet 服务之间增量同步数据的用的，也能用于定时落地数据库数据。谁有写实际的应用例子不？

**Comment by Gowa2017** [comment]
这东西。不走c层是如何实现同步的？

**Reply by hanxi** [reply]
消息推送diff呀

**Reply by Gowa2017** [reply]
。。。消息推送你推送整个消息 还是推送更新的某一部分呢。。

**Reply by hanxi** [reply]
第一次推完整的，之后定时推增量的。

**Reply by Gowa2017** [reply]
> 第一次推完整的，之后定时推增量的。

你就是推送消息，你推送键和数据 而已，然后更新就用了。还需要做点什么工作？

不是很了解。但事实上我觉得简单粗暴的如果要持久化，不若是全量给他弄过去。弄个更新的过去，然后再落地，就是为了减少传输的数据量？
